apps/wpcom-block-editor/src/wpcom/features/live-preview/upgrade-modal.tsx

```diff
@@ -0,0 +1,98 @@
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+import { FC, useEffect, useState } from 'react';
+import { ThemeUpgradeModal } from 'calypso/components/theme-upgrade-modal';
+import { getUnlock } from './utils';
+import './upgrade-modal.scss';
+
+const SAVE_HUB_SAVE_BUTTON_SELECTOR = '.edit-site-save-hub__button';
+const HEADER_SAVE_BUTTON_SELECTOR = '.edit-site-save-button__button';
+
+const GRIDICONS_URL =
+	'https://widgets.wp.com/wpcom-block-editor/images/gridicons-47c7fb356fcb2d963681.svg';
+const GRIDICONS_ID = 'wpcom-live-preview-gridicons';
+
+const unlock = getUnlock();
+
+export const LivePreviewUpgradeModal: FC< { themeId: string; upgradePlan: () => void } > = ( {
+	themeId,
+	upgradePlan,
+} ) => {
+	const [ isThemeUpgradeModalOpen, setIsThemeUpgradeModalOpen ] = useState( false );
+	const canvasMode = useSelect(
+		( select ) =>
+			unlock && select( 'core/edit-site' ) && unlock( select( 'core/edit-site' ) ).getCanvasMode(),
+		[]
+	);
+
+	/**
+	 * This overrides the `SaveButton` behavior by adding a listener and changing the copy.
+	 * Our objective is to open a custom modal ('ThemeUpgradeModal') instead of proceeding with the default behavior.
+	 * For more context, see the discussion on adding an official customization method: https://github.com/WordPress/gutenberg/pull/56807.
+	 */
+	useEffect( () => {
+		const handler: EventListener = ( e ) => {
+			e.stopPropagation();
+			setIsThemeUpgradeModalOpen( true );
+		};
+		const overrideSaveButton = ( selector: string ) => {
+			const button = document.querySelector( selector );
+			if ( button ) {
+				button.textContent = __( 'Upgrade now', 'wpcom-live-preview' );
+				button.addEventListener( 'click', handler );
+			}
+		};
+		if ( canvasMode === 'view' ) {
+			overrideSaveButton( SAVE_HUB_SAVE_BUTTON_SELECTOR );
+			return;
+		}
+		if ( canvasMode === 'edit' ) {
+			overrideSaveButton( HEADER_SAVE_BUTTON_SELECTOR );
+			return;
+		}
+		return () => {
+			document
+				.querySelector( SAVE_HUB_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'click', handler );
+			document
+				.querySelector( HEADER_SAVE_BUTTON_SELECTOR )
+				?.removeEventListener( 'click', handler );
+		};
+	}, [ canvasMode ] );
+
+	/**
+	 * Load the SVG sprite for `Gridicon` in `@automattic/components`.
```

Comment:
I followed the existing code for other apps and could not delve deeper to find a better way. 🤔 

- https://github.com/Automattic/wp-calypso/blob/373449edcf24f6488de2e084b3595b58a8aebfd4/apps/blaze-dashboard/README.md?plain=1#L30-L41
- https://github.com/Automattic/jetpack/blob/5e1e8c83d06ac24ccc88afdcc386c54d61242709/projects/packages/blaze/src/class-dashboard.php#L83-L89
- https://github.com/Automattic/wp-calypso/blob/373449edcf24f6488de2e084b3595b58a8aebfd4/apps/blaze-dashboard/webpack.config.js#L187-L190

Reply:
@okmttdhr We can just inline the SVG code. See: https://github.com/Automattic/wp-calypso/pull/82787

cc: @arthur791004 

Sorry for the late review!

Reply:
Yes, but I was concerned that we might need to discontinue using Gridicons in `client/components/theme-upgrade-modal` (potentially other components within `client/components`). I wanted to keep the component as is as possible. 
But it should be acceptable in this specific case.

Reply:
>  We can just inline the SVG code

If so, we have to modify the `ThemeUpgradeModal` to use the inline SVG but it seems weird as the component is located under the `client/components` folder and it should use `GridIcon` directly 😅

Reply:
I had a try with replacing icons on build time. WDYT?

Reply:
Sorry maybe I'm missing something but why not just replace gridicons with WP icons in that `ThemeUpgradeModal` directly?

Reply:
To be sure, did you see the comment above? 👀 
https://github.com/Automattic/wp-calypso/pull/85013#discussion_r1424813237

Reply:
Yes I did. I mean, I don't see why we can't use `@wordpress/icons` in the `ThemeUpgradeModal` if it renders essentially the same thing (a checkmark).

Reply:
> I don't see why we can't use @wordpress/icons in the ThemeUpgradeModal if it renders essentially the same thing (a checkmark).

+1. I remember I left the same comment somewhere but I cannot see it 😂

Reply:
> How about replacing icons in the modal with the WordPress icon if possible?

Ooops this

Reply:
Thanks, I’ll address it in that way! 👍 

While I agree that it's acceptable in this specific case, I just want to note that my concern was primarily about replacing the original icons (Gridicons), not specifically about whether to implement them inline. https://github.com/Automattic/wp-calypso/pull/85013#discussion_r1424667153

I've now realized that in Calypso, several different icon sets are being used (`@wordpress/icons`, `Gridicons`, `MaterialIcon`). This is not ideal, ![badge](https://img.shields.io/badge/review-imo-orange.svg). Icons are not something interchangeably replaced with different sets even if they appear similar, and are supposed to be integral to our brand identity.
After things have settled on my end, I’ll create an issue to make it visible.

Reply:
📝 p7H4VZ-4JT-p2